master/internal/hpimportance/manager.go

```diff
@@ -0,0 +1,284 @@
+package hpimportance
+
+import (
+	"time"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+type metricType int
+
+// Training designates metrics from training steps.
+const Training metricType = 0
+
+// Validation designates metrics from validation steps.
+const Validation metricType = 1
+
+// Pending indicates that a computation request is queued.
+const Pending string = "pending"
+
+// InProgress indicates that a computation request is in-progress.
+const InProgress string = "in_progress"
+
+// Complete indicates that one request was completed and no further requests have been received.
+const Complete string = "complete"
+
+// Failed indicates that there was an error during computation.
+const Failed string = "failed"
+
+const (
+	// RootAddr is the path to use for looking up the manager actor.
+	RootAddr = "hpimportance"
+
+	// Evaluate after every 10%, but no more than every 10 minutes
+	minPause   = 10 * time.Minute
+	minPercent = 0.1
+)
+
+// TerminalStates indicate final states, as opposed to tasks that imply imminent changes.
+var TerminalStates = map[string]bool{
+	Complete: true,
+	Failed:   true,
+}
+
+// ExperimentCreated is the message an experiment sends when created.
+type ExperimentCreated struct {
+	ID int
+}
+
+// ExperimentCompleted is the message an experiment sends upon completion.
+type ExperimentCompleted struct {
+	ID int
+}
+
+// ExperimentPaused is the message an experiment sends on pausing.
+type ExperimentPaused struct {
+	ID int
+}
+
+// ExperimentProgress is the message an experiment sends after trial completion.
+type ExperimentProgress struct {
+	ID       int
+	Progress float64
+}
+
+// The zero-values of these types happen to be sensible defaults too. If we add fields for which
+// that is not true, add a newStateRecord(). This is for state that doesn't need to be persisted,
+// especially if it is access very frequently.
+type stateRecord struct {
+	lastResult   time.Time
+	lastProgress float64
+}
+
+type manager struct {
+	db    *db.PgDB
+	state map[int]stateRecord
+}
+
+// NewManager initializes the master actor (of which there should only be one instance running).
+func NewManager(db *db.PgDB) actor.Actor {
+	return &manager{db, make(map[int]stateRecord)}
+}
+
+func (m *manager) Receive(ctx *actor.Context) error {
+	switch msg := ctx.Message().(type) {
+	case actor.PreStart:
+		// TODO: fetch any pending or in_progress tasks from the DB and trigger them
```

Comment:
I may do this sooner rather than later, although for now it's moot since the actual HP importance algorithm isn't yet implemented.

Reply:
A JIRA was filed for this.